src/report.rs

```diff
@@ -0,0 +1,149 @@
+        let misc_select = u32::from_le(self.miscselect);
+        MiscSelect::from_bits(misc_select).ok_or(Error::InvalidMiscSelect(misc_select))
+    }
+
+    /// Set of flags describing attributes of the enclave.
+    pub fn attributes(&self) -> Result<Attributes, Error> {
+        let features = u64::from_le(self.features);
+        let features = Features::from_bits(features).ok_or(Error::InvalidFeatures(features))?;
+        let xfrm = u64::from_le(self.xfrm);
+        let xfrm = Xfrm::from_bits(xfrm).ok_or(Error::InvalidXfrm(xfrm))?;
+        Ok(Attributes::new(features, xfrm))
+    }
+
+    /// Value of SECS.MRENCLAVE (measurement of the enclave).
+    pub fn enclave_measurement(&self) -> [u8; 32] {
+        self.mrenclave
+    }
+
+    /// Value from SECS.MRSIGNER (hash of the enclave signing key).
+    pub fn enclave_signing_key_hash(&self) -> [u8; 32] {
+        self.mrsigner
+    }
+
+    /// ISV assigned Product ID of the enclave.
+    pub fn enclave_product_id(&self) -> u16 {
+        u16::from_le(self.isv_prodid)
+    }
+
+    /// ISV assigned SVN (security version number) of the enclave.
+    pub fn enclave_security_version(&self) -> u16 {
+        u16::from_le(self.isv_svn)
+    }
+
+    /// Data provided by the user and protected by the reports MAC.
+    pub fn report_data(&self) -> [u8; 64] {
+        self.reportdata
+    }
+}
+
+/// The REPORT structure is the output of the EREPORT instruction, and must be 512-Byte aligned.
+///
+/// For more information see:
+///
+/// [Intel® 64 and IA-32 Architectures Software Developer's Manual Volume 3 (3A, 3B, 3C & 3D): System Programming Guide]
+/// (https://www.intel.com/content/www/us/en/architecture-and-technology/64-ia-32-architectures-software-developer-vol-3d-part-4-manual.html)
+///
+/// Table 38-21. Layout of REPORT
+#[derive(Clone, Debug)]
+#[repr(C, align(512))]
+pub struct Report {
+    body: ReportBody,
+    keyid: [u8; 32],
+    mac: [u8; 16],
+}
+
+impl Report {
+    /// The enclave report body.
+    pub fn body(&self) -> &ReportBody {
+        &self.body
+    }
+
+    /// Value for key wear-out protection.
+    pub fn key_id(&self) -> [u8; 32] {
+        self.keyid
+    }
+
+    /// The MAC (message authentication code) on the report using report key.
+    pub fn mac(&self) -> [u8; 16] {
+        self.mac
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::{Report, ReportBody};
+    use testaso::testaso;
+
+    testaso! {
+        struct ReportBody: 8, 384 => {
+            cpusvn: 0,
+            miscselect: 16,
+            reserved1: 20,
+            features: 48,
+            xfrm: 56,
+            mrenclave: 64,
+            reserved2: 96,
+            mrsigner: 128,
+            reserved3: 160,
+            isv_prodid: 256,
+            isv_svn: 258,
+            reserved4: 260,
+            reportdata: 320
+        }
+
+        struct Report: 512, 512 => {
+            body: 0,
+            keyid: 384,
+            mac: 416
+        }
```

Comment:
FTR, this is already defined at https://github.com/enarx/sallyport/blob/14eaad9638a22b21da1e4d02273aba6c23666fda/src/item/enarxcall/sgx.rs#L193-L219